runtime/legion/legion_ops.cc

```diff
@@ -5444,7 +5606,7 @@ namespace Legion {
         case MAPPING_FENCE:
           {
             complete_mapping();
-            complete_execution();
+            complete_execution(Runtime::protect_event(execution_fence_event));
```

Comment:
Why is this necessary? Mapping fences should only need to depend on the mapping of prior operations. It should have no dependence on the execution in any way.

Reply:
If that's the intended semantics, I will revert this change.